addons/mrp/models/product.py

```diff
@@ -109,7 +109,12 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
         This override is used to get the correct quantities of products
         with 'phantom' as BoM type.
         """
-        bom_kits = self.env['mrp.bom']._get_product2bom(self, bom_type='phantom')
+        products = self.filtered(lambda p: p.type  != 'service')
+        bom_kits = {
+            product: bom
+            for product, bom in zip(products, self.env['mrp.bom']._get_product2bom(products, bom_type='phantom'))
+            if bom
+        }
```

Comment:
isn't the same as 

        products = self.filtered(lambda p: p.type  != 'service')
        bom_kits = self.env['mrp.bom']._get_product2bom(products, bom_type='phantom')


?￼

Reply:
the previous version of `_get_product2bom` yes, the version introduced by this patch returns either a list of boms, list of `False` or empty recordset which while it's one of Python's strengths, is poor in practice (I'd rather have a single return type if possible)